### PR TITLE
combining kp with no data

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Fix a bug that presented itself when the list of kp_times was empty (nothing was
downloaded).  Will now return a padded list of NaN.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

```
import pysat
import datetime as dt

standard_inst = pysat.Instrument('sw', 'kp', '')
recent_inst = pysat.Instrument('sw', 'kp', 'recent')

stime = dt.datetime(2014, 2, 19)
etime = dt.datetime(2014, 2, 24)

standard_inst.download(start=stime, stop=etime, freq='M')
recent_inst.download(start=stime, stop=etime)

kp_inst = python.instrument.methods.sw.combine_kp(standard_inst=standard_inst, recent_inst=recent_inst, start=stime, stop=etime)

print(kp_inst)
```

If you take out `freq='M'`, you will get data.  With it in, you should get NaN.

**Test Configuration**:
* OS X Sierra
* Python 2.7

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by rstoneback** [comment]
Worked for me, as described above, with a small typo correction to the example combine command,

kp_inst = pysat.instruments.methods.sw.combine_kp(standard_inst=standard_inst, recent_inst=recent_inst, start=stime, stop=etime)

**Comment by rstoneback** [comment]
Looks like a unit test would require Kp data. Travis doesn't support ftp anymore so automated tests of this code aren't currently possible.

File: pysat/instruments/methods/sw.py [hunk 1]
```diff
@@ -193,10 +193,14 @@ def combine_kp(standard_inst=None, recent_inst=None, forecast_inst=None,
         notes += "{:})".format(itime.date())
 
     # Determine if the beginning or end of the time series needs to be padded
-    freq = pysat.utils.time.calc_freq(kp_times)
+
```
Comment (rstoneback) [inline pysat/instruments/methods/sw.py]: blank line

**Comment by jklenzing** [comment]
> Looks like a unit test would require Kp data. Travis doesn't support ftp anymore so automated tests of this code aren't currently possible.

There is some test data in the tests folder for kp.  Was using this for some of the tests before.

**Comment by jklenzing** [comment]
Found it.  Historical data is archived in the test folder and is used as part of the `TestSwKpCombine` class in `test_sw.py`.

**Comment by rstoneback** [comment]
Cool. We should test for this then.

**Comment by jklenzing** [comment]
Actually, looking into the tests, if this is a bug in the download method, we can't test for it..

**Comment by rstoneback** [comment]
The change is to the combine kp method.

**Comment by jklenzing** [comment]
OK, I got this figured out.  Test forthcoming.

**Comment by jklenzing** [comment]
Merged in the travis fixes and added a unit test.

**Comment by jklenzing** [comment]
@aburrell, can you verify this test covers the situation?  Also, will a similar bug be present in combine_f107?

**Comment by aburrell** [comment]
I changed the test so that it worked.  A similar bug is probably present in combine_f107, it's just harder to encounter since the historic data set is so long.

**Pull request merged**